src/main/java/core/basesyntax/service/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.OperationHandler;
+
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy{
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getOperationHandler(FruitTransaction.Operation operation) {
+        return operationHandlers.get(operation);
```

Comment:
Consider throwing a custom exception if the operation is not supported, which means the 'operationHandlers' map does not contain a handler for the provided 'operation'.